ibazel/profiler/profiler.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(){
+  function setOption(options, name, value) {
+    if (typeof value === 'undefined') {
+      return;
+    }
+    if (!isNaN(+value)) {
+      value = +value;
+    }
+    return options[name] = value;
+  };
+
+  function extractOptions() {
+    var element, keyAndValue, m, mm, options, pair, src, _i, _j, _len, _len1, _ref, _ref1;
+    _ref = window.document.getElementsByTagName('script');
+    for (_i = 0, _len = _ref.length; _i < _len; _i++) {
```

Comment:
`.getElementsByTagName` returns a live [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection) which means you can't directly iterate over it with a for of, but you can do

```
[...window.document.getElementsByTagName('script')].forEach(...)
```

which does some of the bookkeeping for you and is a bit cleaner IMO

Reply:
Thanks. That's nice and clean.